fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java

```diff
@@ -70,14 +72,23 @@ public StatsCalculator(GroupExpression groupExpression) {
      * Do estimate.
      */
     public void estimate() {
+
         StatsDeriveResult stats = groupExpression.getPlan().accept(this, null);
         groupExpression.getOwnerGroup().setStatistics(stats);
-        Plan plan = groupExpression.getPlan();
-        long limit = plan.getLimit();
-        if (limit != -1) {
-            stats.setRowCount(Math.min(limit, stats.getRowCount()));
-        }
         groupExpression.setStatDerived(true);
+
+    }
+
+    @Override
+    public StatsDeriveResult visitLogicalLimit(LogicalLimit<Plan> limit, Void context) {
+        StatsDeriveResult stats = groupExpression.getCopyOfChildStats(0);
```

Comment:
both derive logic in LogicalLimit and PhysicalLimit is similar ,I think extract these common logic to a seperate method maybe better